src/stp/invariants/WeHaveBallInvariant.cpp

```diff
@@ -5,8 +5,8 @@
 #include "stp/invariants/WeHaveBallInvariant.h"
 
 namespace rtt::ai::stp::invariant {
-    bool WeHaveBallInvariant::checkInvariant(world_new::view::WorldDataView world, const Field *field) const noexcept {
-        auto& us = world.getUs();
-        return std::any_of(us.begin(), us.end(), [](auto& robot){return robot.hasBall();});
-    }
+uint8_t WeHaveBallInvariant::metricCheck(world_new::view::WorldDataView world, const Field *field) const noexcept {
+    auto& us = world.getUs();
+    return std::any_of(us.begin(), us.end(), [](auto& robot) { return robot.hasBall(); }) ? stp::control_constants::FUZZY_TRUE : stp::control_constants::FUZZY_FALSE;
```

Comment:
I think you are returning FUZZY_TRUE because now the real false and true are taken to be 0 and 255? If that's the case, this is not entirely intuitive and we might be able to find a better solution for this

Reply:
Do you have any idea how this could be improved? Basically this statement returns 255(fuzzy_true) if any of the robots has the ball and 0(fuzzy_false) if none have the ball.

Reply:
Yeah it's fine functionally, but it might be a bit unclear to people. However, we can fix this later, I'll make it an issue. It is still understandable